sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -431,11 +432,63 @@ public void canDeleteNetworkWithServiceCallBack() {
         Assertions.assertEquals(counter.intValue(), 1);
     }
 
+    @Test
+    public void canSetSubnetAddressPrefixes() {
+        String networkName = generateRandomResourceName("vnet", 10);
+        String subnetName = "subnet1";
+        // create withAddressPrefix
+        Network network = networkManager.networks()
+            .define(networkName)
+            .withRegion(Region.US_EAST)
+            .withNewResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/24")
+            .withSubnet(subnetName, "10.0.0.0/29")
+            .create();
+
+        Subnet subnet = network.subnets().get(subnetName);
+        Assertions.assertNotNull(subnet.addressPrefix());
+        Assertions.assertEquals(1, subnet.addressPrefixes().size());
+        // test service behavior, they will return either addressPrefix or addressPrefixes, not both
+        Assertions.assertNotNull(subnet.innerModel().addressPrefix());
+        Assertions.assertTrue(CoreUtils.isNullOrEmpty(subnet.innerModel().addressPrefixes()));
+
+        // update withAddressPrefixes
+        network.update().updateSubnet(subnetName)
+            .withAddressPrefixes(Arrays.asList("10.0.0.8/29", "10.0.0.16/29"))
+            .parent()
+            .apply();
+
+        network.refresh();
+        subnet = network.subnets().get(subnetName);
+
+        Assertions.assertEquals(2, subnet.addressPrefixes().size());
+        Assertions.assertTrue(subnet.addressPrefixes().contains("10.0.0.8/29"));
+        Assertions.assertNotNull(subnet.addressPrefix());
+        // test service behavior, they will return either addressPrefix or addressPrefixes, not both
+        Assertions.assertEquals(2, subnet.innerModel().addressPrefixes().size());
+        Assertions.assertNull(subnet.innerModel().addressPrefix());
+
+        // update withAddressPrefix
+        network.update().updateSubnet(subnetName)
+            .withAddressPrefix("10.0.0.0/29")
+            .parent()
+            .apply();
+
+        network.refresh();
+        subnet = network.subnets().get(subnetName);
+
+        Assertions.assertEquals("10.0.0.0/29", subnet.addressPrefix());
+        Assertions.assertEquals(1, subnet.addressPrefixes().size());
+        // test service behavior, they will return either addressPrefix or addressPrefixes, not both
+        Assertions.assertNotNull(subnet.innerModel().addressPrefix());
+        Assertions.assertTrue(CoreUtils.isNullOrEmpty(subnet.innerModel().addressPrefixes()));
```

Comment:
BTW, if we `listAvailablePrivateIPAddresses()` on the subnet, would it return a list of single IP address (the one we send in query, and it is available), or a list of IP addresses (any available IPs which may include the one we send)

Reply:
Seems any available IPs(not restrained to any one of the addressPrefixes):
For a subnet `.withAddressPrefixes(Arrays.asList("10.0.0.8/29", "10.0.0.16/29"))`

1st call:
> vnet74253/CheckIPAddressAvailability?ipAddress=10.0.0.8

result:
```json
"ResponseBody": {
  "available": false,
  "availableIPAddresses": [
    "10.0.0.12",
    "10.0.0.13",
    "10.0.0.14",
    "10.0.0.20",
    "10.0.0.21"
  ],
  "isPlatformReserved": true
}
```

when we occupy all of them, then do a 2nd call:
> vnet74253/CheckIPAddressAvailability?ipAddress=10.0.0.8

result:
```json
"ResponseBody": {
  "available": false,
  "availableIPAddresses": [
    "10.0.0.22"
  ],
  "isPlatformReserved": true
}
```

This `10.0.0.22` can also be occupied. Seems not a deterministic behavior...

Reply:
This `available` seems to indicate the availability of the `startIp` that we pass in, same as the `isPlatformReserved` property.

Reply:
If we do with an actually available IP, e.g.
> vnet74253/CheckIPAddressAvailability?ipAddress=10.0.0.21

(or 10.0.0.22)
result would be 
```json
"ResponseBody": {
  "available": true,
  "availableIPAddresses": null,
  "isPlatformReserved": false
}
```

Reply:
Yeah, maybe fine.

Let's at least add a comment about the behavior, and the reason why we didn't include it (backward compatible + gateway ?).

Reply:
> Seems any available IPs(not restrained to any one of the addressPrefixes): For a subnet `.withAddressPrefixes(Arrays.asList("10.0.0.8/29", "10.0.0.16/29"))`
> 
> 1st call:
> 
> > vnet74253/CheckIPAddressAvailability?ipAddress=10.0.0.8
> 
> result:
> 
> ```json
> "ResponseBody": {
>   "available": false,
>   "availableIPAddresses": [
>     "10.0.0.12",
>     "10.0.0.13",
>     "10.0.0.14",
>     "10.0.0.20",
>     "10.0.0.21"
>   ],
>   "isPlatformReserved": true
> }
> ```
> 
> when we occupy all of them, then do a 2nd call:
> 
> > vnet74253/CheckIPAddressAvailability?ipAddress=10.0.0.8
> 
> result:
> 
> ```json
> "ResponseBody": {
>   "available": false,
>   "availableIPAddresses": [
>     "10.0.0.22"
>   ],
>   "isPlatformReserved": true
> }
> ```
> 
> This `10.0.0.22` can also be occupied. Seems not a deterministic behavior...

Yeah, I guess it is not determinstic behavior. I guess backend just find a number of available IPs (via some logic) and return. It won't find all of them (and it is probably not possible/necessary if the space is large).